packages/strapi-plugin-i18n/config/policies/validateLocaleCreation.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const { get } = require('lodash/fp');
+const { getService } = require('../../utils');
+
+const validateLocaleCreation = async (ctx, next) => {
+  const { model } = ctx.params;
+  const { query, body } = ctx.request;
+
+  const ctService = getService('content-types');
+
+  if (ctService.isLocalized(model)) {
+    const locale = get('plugins.i18n.locale', query);
+    const relatedEntityId = get('plugins.i18n.relatedEntityId', query);
+
+    try {
+      await ctService.addLocale(body, locale);
+    } catch (e) {
+      return ctx.badRequest("This locale doesn't exist");
+    }
+
+    try {
+      await ctService.addLocalizations(body, { relatedEntityId, model, locale: body.locale });
```

Comment:
can you split the logic between getting the localizations & setting it on the body. 
THis will allow to reuse the first part for the enduser api